CMake/install_codesign_script_OpenStudioApp.cmake

```diff
@@ -0,0 +1,307 @@
+
+This script will therefore run in the CPack staging area, and should be added after installation of root dylibs (they may come from fixup_executable)
+after any rpath adjustments, to ensure the signature sticks.
+
+Usage::
+
+  if(APPLE AND CPACK_CODESIGNING_DEVELOPPER_ID_APPLICATION)
+    install(CODE "set(CPACK_CODESIGNING_DEVELOPPER_ID_APPLICATION \"${CPACK_CODESIGNING_DEVELOPPER_ID_APPLICATION}\")" COMPONENT OpenStudioApp)
+    install(CODE "set(CPACK_CODESIGNING_MACOS_IDENTIFIER \"${CPACK_CODESIGNING_MACOS_IDENTIFIER}\")" COMPONENT OpenStudioApp)
+    # call the script
+    install(SCRIPT "${CMAKE_CURRENT_LIST_DIR}/install_codesign_script_OpenStudioApp.cmake" COMPONENT OpenStudioApp)
+  endif()
+#]=======================================================================]
+
+#------------------------------------------------------------------------------
+# Just a helper
+function(print_relative_paths)
+  set(prefix "")
+  set(valueLessKeywords NAME_ONLY NEWLINE)
+  set(singleValueKeywords PREFIX BASE_DIRECTORY)
+  set(multiValueKeywords ABSOLUTE_PATHS)
+  cmake_parse_arguments(
+    PARSE_ARGV 0 # Start at one with NAME is the first param
+      "${prefix}"
+      "${valueLessKeywords}"
+      "${singleValueKeywords}"
+      "${multiValueKeywords}"
+  )
+
+  if (_UNPARSED_ARGUMENTS)
+    message(FATAL_ERROR "Extra unknown arguments were passed: ${_UNPARSED_ARGUMENTS}")
+  endif()
+  if (_KEYWORDS_MISSING_VALUES)
+    message(FATAL_ERROR "Keywords missing values: ${_KEYWORDS_MISSING_VALUES}")
+  endif()
+
+  if(NOT _BASE_DIRECTORY)
+    set(_BASE_DIRECTORY ${CMAKE_INSTALL_PREFIX})
+  endif()
+
+  foreach(path ${_ABSOLUTE_PATHS})
+    if(_NAME_ONLY)
+      cmake_path(GET path FILENAME rel_path)
+    else()
+      cmake_path(RELATIVE_PATH path BASE_DIRECTORY ${_BASE_DIRECTORY} OUTPUT_VARIABLE rel_path)
+    endif()
+    list(APPEND rel_paths ${rel_path})
+  endforeach()
+
+  if(_NEWLINE)
+    message("${_PREFIX}")
+    foreach(path ${rel_paths})
+      message("   - ${path}")
+    endforeach()
+  else()
+    message("${_PREFIX}${rel_paths}")
+  endif()
+endfunction()
+#------------------------------------------------------------------------------
+
+message("Codesigning inner executables and library from ${CMAKE_CURRENT_LIST_FILE}")
+
+message("CMAKE_INSTALL_PREFIX=${CMAKE_INSTALL_PREFIX}")
+message("CPACK_CODESIGNING_DEVELOPPER_ID_APPLICATION=${CPACK_CODESIGNING_DEVELOPPER_ID_APPLICATION}")
+message("CPACK_CODESIGNING_MACOS_IDENTIFIER=${CPACK_CODESIGNING_MACOS_IDENTIFIER}")
+message("CMAKE_INSTALL_COMPONENT=${CMAKE_INSTALL_COMPONENT}")
+message("CPACK_GENERATOR=${CPACK_GENERATOR}")
+
+if(NOT CPACK_CODESIGNING_DEVELOPPER_ID_APPLICATION)
+  message(FATAL_ERROR "CPACK_CODESIGNING_DEVELOPPER_ID_APPLICATION is required")
+endif()
+
+if(NOT CPACK_CODESIGNING_MACOS_IDENTIFIER)
+  message(FATAL_ERROR "CPACK_CODESIGNING_MACOS_IDENTIFIER is required")
+endif()
+
+
+set(FILES_TO_SIGN
```

Comment:
Is there anyway to get this list of executables from CMake?  Or do a glob after the build?

Reply:
Yes. But that is complicated enough for now and will work for the next 6months.

I'm not sure if I  can do pure cmake glob and find executables or libraries, but we could make a python script similar to the verify_signature I created and call that from cmake.
We'll need to have python as a build dependency but since we use conan already that's not a problem

Reply:
Will work until the next time we upgrade os sdk to be clear. And then the signature testing I set on CI will tell us when it breaks and which files aren't signed properly